### PR TITLE
Twitter integration

### PR DESCRIPTION
**Pull request by yramanchuk**
- Implemented Twitter Provider for Firebase Auth UI
- Implemented Obj-C demo for Twiter Provider

### CODE REVIEW

File: FirebaseTwitterAuthUI/FIRTwitterAuthUI.m [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+//  limitations under the License.
+//
+
+#import <FirebaseAuth/FIRTwitterAuthProvider.h>
+#import <FirebaseAuthUI/FIRAuthUIErrorUtils.h>
+#import <TwitterKit/TwitterKit.h>
+#import "FIRTwitterAuthUI.h"
+
+/** @var kBundleFileName
+ @brief The name of the bundle containing Facebook auth provider assets/resources.
```
Comment (morganchen12) [inline FirebaseTwitterAuthUI/FIRTwitterAuthUI.m]: Change `Facebook` to `Twitter` in comment

File: FirebaseTwitterAuthUI/FIRTwitterAuthUI.m [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+ */
+static NSString *const kBundleFileName = @"FirebaseTwitterAuthUIBundle.bundle";
+
+/** @var kTableName
+ @brief The name of the strings table to search for localized strings.
+ */
+static NSString *const kTableName = @"FirebaseTwitterAuthUI";
+
+/** @var kSignInWithFacebook
```
Comment (morganchen12) [inline FirebaseTwitterAuthUI/FIRTwitterAuthUI.m]: change to `kSignInWithTwitter`

**Comment by morganchen12** [comment]
Looks like travis ci isn't running `pod repo update` and is failing because it can't find TwitterKit :/

File: FirebaseTwitterAuthUI/FIRTwitterAuthUI.m [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                                            secret:session.authTokenSecret];
+       if (completion) {
+         completion(credential, error);
+       }
+     } else {
+       NSError *newError =
+       [FIRAuthUIErrorUtils providerErrorWithUnderlyingError:error
+                                                  providerID:FIRTwitterAuthProviderID];
+       if (completion) {
+         completion(nil, error);
```
Comment (morganchen12) [inline FirebaseTwitterAuthUI/FIRTwitterAuthUI.m]: Should this be `completion(nil, newError)`?

Comment (yramanchuk) [inline FirebaseTwitterAuthUI/FIRTwitterAuthUI.m]: fixed in both cases:
`     if (session) {
       FIRAuthCredential *credential =
       [FIRTwitterAuthProvider credentialWithToken:session.authToken
                                            secret:session.authTokenSecret];
       if (completion) {
         completion(credential, nil);
       }
     } else {
       NSError *newError =
       [FIRAuthUIErrorUtils providerErrorWithUnderlyingError:error
                                                  providerID:FIRTwitterAuthProviderID];
       if (completion) {
         completion(nil, newError);
       }
     }
`

File: FirebaseTwitterAuthUI/FirebaseTwitterAuthUI.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+#import <UIKit/UIKit.h>
+
+//! Project version number for FirebaseTwitterAuthUI.
+FOUNDATION_EXPORT double FirebaseTwitterAuthUIVersionNumber;
+
+//! Project version string for FirebaseTwitterAuthUI.
+FOUNDATION_EXPORT const unsigned char FirebaseTwitterAuthUIVersionString[];
+
+#import <FirebaseTwitterAuthUI/FIRTwitterAuthUI.h>
```
Comment (morganchen12) [inline FirebaseTwitterAuthUI/FirebaseTwitterAuthUI.h]: `FIRTwitterAuthUI.h` needs to be marked as a Public header (not Project header) or it'll cause build errors when integrating as a framework.

Comment (morganchen12) [inline FirebaseTwitterAuthUI/FirebaseTwitterAuthUI.h]: screen shot 2016-09-15 at 12 15 02 pm

File: FirebaseUI.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -7,6 +7,14 @@
 	objects = {
```
Comment (morganchen12) [inline FirebaseUI.xcodeproj/project.pbxproj]: The `FirebaseTwitterAuthUI` target needs to have a target dependency on `FirebaseAuthUI` or it won't compile.
screen shot 2016-09-15 at 12 37 31 pm

File: FirebaseTwitterAuthUI/FIRTwitterAuthUI.m (hunk 1 above)
Comment (yramanchuk) [inline FirebaseTwitterAuthUI/FIRTwitterAuthUI.m]: Done

File: FirebaseTwitterAuthUI/FIRTwitterAuthUI.m (hunk 2 above)
Comment (yramanchuk) [inline FirebaseTwitterAuthUI/FIRTwitterAuthUI.m]: Done!

File: FirebaseUI.xcodeproj/project.pbxproj (hunk 5 above)
Comment (yramanchuk) [inline FirebaseUI.xcodeproj/project.pbxproj]: Done!

**Pull request merged**